### DISCUSSION TITLE
Does't work well with multi-cursor mode

### DISCUSSION THREAD

**Discussion by moritzWa** [community/community#15656] [Copilot Conversations]
**Problem**: When editing code with multiple cursors, co-pilot only predicts the text completion for **ONE** cursor instance. When one then types forward arrow (or tab, depending on one's settings), co-pilot pasts the suggestion for only one cursor. 
Example: 
<img width="410" alt="image" src="https://user-images.githubusercontent.com/42035131/165420192-85ff5240-3587-4982-901a-26e174eb24d2.png"> 
<img width="419" alt="image" src="https://user-images.githubusercontent.com/42035131/165420244-928de128-b5bb-4ca8-9640-5fdd69838c09.png">
 (note that only of the three cursors will have a co-pilot suggestion)

**Solutions**:
a) Make co-pilot work for all cursors (and just copy the prediction for each cursor?)
b) Disable co-pilot suggestions for multiple-cursor mode.

**Comment by harrypuuter** [comment]
Any news on this? Support for multiple cursors is still lacking.

**Reply by ggoonnzzaallo** [reply]
@harrypuuter I posted below after waiting over a year for a solution to this, I just end up building my own extension to handle multi-cursor autocompletes. You can try it [here](https://marketplace.visualstudio.com/items?itemName=doublebot.doublebot) if interested.

**Comment by DavidFarago** [comment]
Such a fix would be awesome, because predictions breaking your multi-cursor edits is very annoying. I searched for quite some time to find this bug report -- I was sure there was one for this annoying behavior. Just disabling co-pilot suggestions (solution b above) should be an easy fix and would hinder my spinal cord to hit tab upon seeing some ghost text.

**Comment by elkir** [comment]
I also reported this but accidentally to the Copilot Chat issue tracker. Here are some more examples to illustrate : 

![image](https://github.com/microsoft/vscode-copilot-release/assets/20708119/b20625eb-7c74-48eb-93ba-6926449ebb69)

As seen in the picture above, the copilot suggestions are only applied to the first cursor of the multicursor.

Normally, multiple cursors are used to input the same text at multiple locations, so it would make sense to apply the suggestion across all of them. There is, of course, the caveat that the suggestion is tailored only to the first line and might not produce useful results for the consequent lines, and future multi-suggestion approach might be useful, but for now, just replicating the suggestion would be sufficient.

The current behaviour produces suggestions that go directly against the workflow of multicursor. The user has to either:
1) continue writing manually ignoring the suggestion (defeating the purpose of Copilot)
![image](https://github.com/microsoft/vscode-copilot-release/assets/20708119/a44a884f-cd69-4621-886d-5723035453e6)

2) if accepted require the user to abandon multi-cursor to continue writing the rest of the instances (defeating the purpose of the multicursor selection)
![image](https://github.com/microsoft/vscode-copilot-release/assets/20708119/226808d2-950b-48a4-a884-c84632882fc0)
![image](https://github.com/microsoft/vscode-copilot-release/assets/20708119/c05ba358-bf60-4828-8e2a-392eeff5c3ed)
![image](https://github.com/microsoft/vscode-copilot-release/assets/20708119/d40f3ee0-5c3d-49c8-b51f-c51e083ed0b0)

**Reply by ggoonnzzaallo** [reply]
I was surprised to see no reply from Microsoft after 1 year+ from this issue so I built my own Copilot extension that's able to handle multi-cursor mode. Attached some examples with your specific code, it basically populated all the lines you've selected after you accept the completion. You can try it [here](https://marketplace.visualstudio.com/items?itemName=doublebot.doublebot) if interested.

![Screenshot 2024-02-28 at 5 10 32 PM](https://github.com/community/community/assets/121972034/d342d99f-a402-4c99-8503-90e34620c9e9)
![Screenshot 2024-02-28 at 5 10 44 PM](https://github.com/community/community/assets/121972034/f05e9244-4b0a-4268-b1ea-7f5ed9838454)

**Reply by tendiesfortwo** [reply]
@ggoonnzzaallo looks cool, will check it out

**Comment by davecheney** [comment]
Duplicate of https://github.com/orgs/community/discussions/7942
